pkg/controller/traceflow/controller.go

```diff
@@ -224,14 +241,27 @@ func (c *Controller) checkTraceflowStatus(tf *opsv1alpha1.Traceflow) (retry bool
 	retry = false
 	sender := false
 	receiver := false
-	for _, nodeResult := range tf.Status.Results {
-		for _, ob := range nodeResult.Observations {
+	for i, nodeResult := range tf.Status.Results {
+		for j, ob := range nodeResult.Observations {
 			if ob.Component == opsv1alpha1.SpoofGuard {
 				sender = true
 			}
 			if ob.Action == opsv1alpha1.Delivered || ob.Action == opsv1alpha1.Dropped {
 				receiver = true
 			}
+			if ob.TranslatedDstIP != "" {
+				pods, err := c.podInformer.Informer().GetIndexer().ByIndex("podIP", ob.TranslatedDstIP)
+				if err != nil {
+					klog.Warningf("Get Pods by IP failed, error: %+v", err)
+				} else if len(pods) > 0 {
+					pod, ok := pods[0].(*corev1.Pod)
+					if !ok {
+						klog.Warningf("Invalid Pod obj in cache")
+					} else {
+						tf.Status.Results[i].Observations[j].Pod = fmt.Sprintf("%s/%s", pod.Namespace, pod.Name)
+					}
+				}
```

Comment:
do we need a log message if no Pods where found with the provided IP address?

Reply:
No, the TranslatedDstIP (Service Endpoint IP) may not be a Pod IP.

Reply:
makes sense, do you think you can add a comment for this?

Reply:
Added comments here
// Add Pod ns/name to observation if TranslatedDstIP (a.k.a. Service Endpoint address) is Pod IP.